### PR TITLE
Mailer Configuration (#1)

### PR DESCRIPTION
**Pull request by rosafaria**
Initial configuration of the mailer and root path.

### CODE REVIEW

**Comment by cweyer** [comment]
Thank you for your contribution, looks good so far. I have some questions though:

* Why did you remove all the documentation?
* Coveralls should say 100% coverage here, don't know why it doesn't but it's in their dasboard, so ok

Regarding coding syle:

* Please make sure to use consistent and correct indentation with soft tabs (2 or 4, depending on the rest of the project)
* Please use consistent string quotes, either with ' or with ". I prefer " for strings I want to use ruby code in with #{} or have control chars like \n and ' otherwise (see http://stackoverflow.com/questions/20870375/single-vs-double-quotes)

File: lib/lotus/mailer.rb [hunk 1]
```diff
@@ -1,7 +1,47 @@
...
+    #
+    # @since 0.1.0
+    # @api private
+    def self.load!
+      configuration.load!
+    end
+    
+    # Reset the configuration
+    def self.reset!
+	  configuration.reset!
```
Comment (joneslee85) [inline lib/lotus/mailer.rb]: nitpick, please use 2 white spaces tab

File: lib/lotus/mailer.rb [hunk 2]
```diff
@@ -1,7 +1,47 @@
-require "lotus/mailer/version"
+require 'set'
+require 'pathname'
```
Comment (joneslee85) [inline lib/lotus/mailer.rb]: If we do not use this lib in this file, do you think we should remove it?

File: lib/lotus/mailer/configuration.rb [hunk 3]
```diff
@@ -0,0 +1,85 @@
+require 'set'
```
Comment (joneslee85) [inline lib/lotus/mailer/configuration.rb]: If we do not use this lib in this file, do you think we should remove it?

File: test/configuration_test.rb [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+  describe '#reset!' do
+    before do
+      @configuration.root 'test'
+      @configuration.reset!
+    end
+
+    it 'resets root' do
+      root = Pathname.new('.').realpath
+      @configuration.root.must_equal         root
```
Comment (joneslee85) [inline test/configuration_test.rb]: can we remove the white spaces in front of `root`?

File: lib/lotus/mailer/configuration.rb [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      #
+      # @example Setting the value
+      #   require 'lotus/mailer'
+      #
+      #   Lotus::Mailer.configure do
+      #     root '/path/to/templates'
+      #   end
+      #
+      #   Lotus::Mailer.configuration.root # => #<Pathname:/path/to/templates>
+
```
Comment (joneslee85) [inline lib/lotus/mailer/configuration.rb]: remove empty ;ine

File: lib/lotus/mailer/configuration.rb (hunk 3 above)
Comment (joneslee85) [inline lib/lotus/mailer/configuration.rb]: Please add `require 'pathname'` here

**Comment by joneslee85** [comment]
:+1:

**Comment by joneslee85** [comment]
@rosafaria please squash all commits into 1 commit before merging

**Comment by rosafaria** [comment]
@joneslee85 I squashed them, please verify if it is correct, since I have never done this before.

**Comment by joneslee85** [comment]
good to merge

**Comment by joneslee85** [comment]
@rosafaria you can merge the PR yourself

**Comment by rosafaria** [comment]
screenshot from 2015-08-04 10 56 04

**Comment by rosafaria** [comment]
That's the message I get...

**Pull request merged**